libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -284,7 +285,7 @@ private void startForegroundNotification(Notification notification, int notifica
    * performing navigation progress on a accurate/valid rawLocation update.
    */
   @SuppressWarnings("MissingPermission")
-  private boolean validLocationUpdate(Location location) {
+  private boolean isValidLocationUpdate(Location location) {
     if (locationEngine.getLastLocation() == null) {
```

Comment:
I think this check should be removed.

If you 👀 the code carefully we're always checking if `location != null` before calling `isValidLocationUpdate` so we know for sure that the location is not `null` at that point. Even more, if the `location` received https://github.com/mapbox/mapbox-navigation-android/pull/756/files#diff-7e57c49bef82db11f00cfa982b5e2e07R99 is not `null` but the `locationEngine.getLastLocation() == null` we're not checking if the location is the same as previous. It that what we want?

Reply:
Yea I think this method should be removed. I think at some point this method was useful but I don't think it really is just for a null check, especially if it's redundant. If you look at the comments, it says it `runs several checks`.